protocols/ping/src/handler.rs

```diff
@@ -103,25 +103,6 @@ impl Config {
         self.max_failures = n;
         self
     }
-
-    /// Sets whether the ping protocol itself should keep the connection alive,
-    /// apart from the maximum allowed failures.
-    ///
-    /// By default, the ping protocol itself allows the connection to be closed
-    /// at any time, i.e. in the absence of ping failures the connection lifetime
-    /// is determined by other protocol handlers.
-    ///
-    /// If the maximum number of allowed ping failures is reached, the
-    /// connection is always terminated as a result of [`ConnectionHandler::poll`]
-    /// returning an error, regardless of the keep-alive setting.
-    #[deprecated(
-        since = "0.40.0",
-        note = "Use `libp2p::swarm::behaviour::KeepAlive` if you need to keep connections alive unconditionally."
-    )]
-    pub fn with_keep_alive(mut self, b: bool) -> Self {
-        self.keep_alive = b;
```

Comment:
What about the field? Can we remove that too?

Reply:
It get's read in `ConnectionHandler::connection_keep_alive`. We could probably just remove it though and always return that it shouldn't be kept alive. As one can't set it anymore anyway.